api/src/routers/meals.js

```diff
@@ -0,0 +1,150 @@
+import express from "express";
+import knex from "../database_client.js";
+
+const mealsRouter = express.Router();
+
+mealsRouter.get("/", async (req, res, next) => {
+  try {
+    const query = knex("meal");
+    const {
+      maxPrice,
+      availableReservations,
+      title,
+      dateAfter,
+      dateBefore,
+      limit,
+      sortKey,
+      sortDir,
+    } = req.query;
+
+    if (maxPrice !== undefined) {
+      query.where("price", "<", maxPrice);
+    }
+    if (availableReservations !== undefined) {
+      if (availableReservations === "true") {
+        //might be returned as a string, hence checking explicitly
+        query
+          .leftJoin("reservation", "meal.id", "=", "reservation.meal_id")
+          .select("meal.id", "meal.max_reservations", "meal.title")
+          .sum("reservation.number_of_guests as sum_of_guests")
+          .groupBy("meal.id", "meal.max_reservations", "meal.title")
+          .having("sum_of_guests", "<", knex.ref("meal.max_reservations"));
+      } else {
+        query
+          .leftJoin("reservation", "meal.id", "=", "reservation.meal_id")
+          .select("meal.id", "meal.max_reservations", "meal.title")
+          .sum("reservation.number_of_guests as sum_of_guests")
+          .groupBy("meal.id", "meal.max_reservations", "meal.title")
+          .having("sum_of_guests", ">=", knex.ref("meal.max_reservations"));
+      }
```

Comment:
these two queries are almost the same. You can merge the two by using ternary conditional operator in the having `condition?ifTrue:ifFalse`